### PR TITLE
Validate service IDs are unique

### PR DESCRIPTION
**Pull request by reinkrul**
Required check for consumers according to https://www.w3.org/TR/did-core/#services

### CODE REVIEW

File: did/validator.go [hunk 1]
```diff
@@ -182,6 +185,12 @@ func (s serviceValidator) Validate(document Document) error {
 		default:
 			return makeValidationError(ErrInvalidService)
 		}
+		// service ID must be unique
+		for x, other := range document.Service {
+			if i != x && service.ID == other.ID {
```
Comment (gerardsn) [inline did/validator.go:190]: small optimization? (if it doesn't copy the sub slice)
```suggestion
		for _, other := range document.Service[i+1:] {
			if service.ID == other.ID {
```

**Review by gerardsn — APPROVED** [review]

File: did/validator.go (hunk 1 above)
Comment (reinkrul) [reply did/validator.go:190]: Neat idea. I think slices of slices in Go are really cheap, since they are basically pointers to the other slice.

**Comment by reinkrul** [comment]
Before merging, we need to check on Nuts dev, stable, test and prd networks whether there are DID documents with duplicate service IDs. Shouldn't be possible (since the ID is a hash of the endpoint) but it could be.

**Comment by woutslakhorst** [comment]
> Before merging, we need to check on Nuts dev, stable, test and prd networks whether there are DID documents with duplicate service IDs. Shouldn't be possible (since the ID is a hash of the endpoint) but it could be.

Possible disastrous consequences, any benefits apart from following the specs?

**Comment by reinkrul** [comment]
Probably no added benefits for now.
